pkg/operator/podidentity/awspodidentitywebhook.go

```diff
@@ -32,3 +39,19 @@ func (a AwsPodIdentity) GetImagePullSpec() string {
 func (a AwsPodIdentity) Name() string {
 	return "aws"
 }
+
+func (a AwsPodIdentity) ApplyDeploymentSubstitutionsInPlace(deployment *appsv1.Deployment, client client.Client, logger log.FieldLogger) error {
+	region, err := awsutils.LoadInfrastructureRegion(client, logger)
+	if err != nil {
+		return err
+	}
+
+	for i := range deployment.Spec.Template.Spec.Containers[0].Args {
+		if strings.Contains(deployment.Spec.Template.Spec.Containers[0].Args[i], "--aws-default-region") {
+			deployment.Spec.Template.Spec.Containers[0].Args[i] = fmt.Sprintf("--aws-default-region=%s", region)
+		}
+	}
+
+	deployment.Spec.Template.Spec.Containers[0].Image = a.GetImagePullSpec()
```

Comment:
This is required for all platforms. Keep this in the controller code to reduce duplicate code. Keep these platform specific functions for platform specific changes.

Reply:
moved it back to controller.